Unknown file extension "" for /usr/lib/node_modules/mscx2ly/bin/mscx2ly

- this is the error I got when I tried to run mscx2ly from command line. Installed it using sudo.
Here's the input and ouput:

> [user]@[machine]:~/[folder]$ mscx2ly --help
> internal/process/esm_loader.js:74
>     internalBinding('errors').triggerUncaughtException(
>                               ^
> 
> TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension "" for /usr/lib/node_modules/mscx2ly/bin/mscx2ly
>     at Loader.defaultGetFormat [as _getFormat] (internal/modules/esm/get_format.js:65:15)
>     at Loader.getFormat (internal/modules/esm/loader.js:116:42)
>     at Loader.getModuleJob (internal/modules/esm/loader.js:247:31)
>     at async Loader.import (internal/modules/esm/loader.js:181:17)
>     at async Object.loadESM (internal/process/esm_loader.js:68:5) {
>   code: 'ERR_UNKNOWN_FILE_EXTENSION'
> }
> 
What is wrong?

OK, 

> node --version
> v12.22.9

install with:

> sudo npm install mscx2ly --global

re-install with https address goes wrong, like this:

> sudo npm install https://github.com/mauritslamers/mscx2ly#ml/es6-script --global
> npm ERR! code 128
> npm ERR! An unknown git error occurred
> npm ERR! command git --no-replace-objects ls-remote ssh://git@github.com/mauritslamers/mscx2ly.git
> npm ERR! Warning: Permanently added 'github.com' (ED25519) to the list of known hosts.
> npm ERR! git@github.com: Permission denied (publickey).
> npm ERR! fatal: Could not read from remote repository.
> npm ERR! 
> npm ERR! Please make sure you have the correct access rights
> npm ERR! and the repository exists.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /root/.npm/_logs/2025-12-10T22_38_38_214Z-debug-0.log
>

It still gives the same error:

> sudo npm install github:mauritslamers/mscx2ly#ml/es6-script --global
> npm ERR! code 128
> npm ERR! An unknown git error occurred
> npm ERR! command git --no-replace-objects ls-remote ssh://git@github.com/mauritslamers/mscx2ly.git
> npm ERR! git@github.com: Permission denied (publickey).
> npm ERR! fatal: Could not read from remote repository.
> npm ERR! 
> npm ERR! Please make sure you have the correct access rights
> npm ERR! and the repository exists.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /root/.npm/_logs/2025-12-11T13_01_03_065Z-debug-0.log
>
The log contains the same error text, but I'll attach the log anyway
file:///root/.npm/_logs/2025-12-11T13_01_03_065Z-debug-0.log

I still get this:

> npm ERR! code 128
> npm ERR! An unknown git error occurred
> npm ERR! command git --no-replace-objects ls-remote ssh://git@github.com/mauritslamers/mscx2ly.git
> npm ERR! git@github.com: Permission denied (publickey).
> npm ERR! fatal: Could not read from remote repository.
> npm ERR! 
> npm ERR! Please make sure you have the correct access rights
> npm ERR! and the repository exists.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /root/.npm/_logs/2025-12-12T07_31_29_667Z-debug-0.log

Is it just a permissions error? Have other people been able to install this without problems?

Sorry, but did you try to rerun the install through the github link, or as
npm install mscx2ly --global? I released a new version, please try the
latter.

Op vr 12 dec 2025, 08:33 schreef juhaem ***@***.***>:

> *juhaem* left a comment (mauritslamers/mscx2ly#1)
> <https://github.com/mauritslamers/mscx2ly/issues/1#issuecomment-3645273231>
>
> I still get this:
>
> npm ERR! code 128
> npm ERR! An unknown git error occurred
> npm ERR! command git --no-replace-objects ls-remote ***@***.***
> /mauritslamers/mscx2ly.git
> npm ERR! ***@***.***: Permission denied (publickey).
> npm ERR! fatal: Could not read from remote repository.
> npm ERR!
> npm ERR! Please make sure you have the correct access rights
> npm ERR! and the repository exists.
>
> npm ERR! A complete log of this run can be found in:
> npm ERR! /root/.npm/_logs/2025-12-12T07_31_29_667Z-debug-0.log
>
> Is it just a permissions error? Have other people been able to install
> this without problems?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mauritslamers/mscx2ly/issues/1#issuecomment-3645273231>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAADSGBLROE7NEMR3CCEMXL4BJVUBAVCNFSM6AAAAACOVG5RTKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTMNBVGI3TGMRTGE>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>

--- a/package.json
+++ b/package.json
@@ -18,6 +18,7 @@
     "musescore"
   ],
   "type": "module",
+  "engines": { "node": ">=18" },
   "typings": "./typings/mscx2ly.d.ts",
   "author": "Maurits Lamers",
   "license": "MIT",

